File: dummy-ai-client/NetworkStandard/Messages/game_operation_message.py
from NetworkStandard.MessageContainer.message_container import MessageContainer
from NetworkStandard.MessageTypeEnum.messagetype_enum import MessageTypeEnum
from NetworkStandard.DataTypes.Operations.operation import Operation

import uuid


class GameOperationMessage(MessageContainer):

    def __init__(self, player_Id, type_enum, creation_date, debug_message,
                 operation):
        super().__init__(player_Id, type_enum, creation_date, debug_message)
        self.operation = operation

File: frontend/core/src/com/sopra/ntts/model/NetworkStandard/Messages/GameOperationMessage.java
package com.sopra.ntts.model.NetworkStandard.Messages;

import com.sopra.ntts.model.NetworkStandard.DataTypes.Operations.Operation;
import com.sopra.ntts.model.NetworkStandard.MessageContainer.MessageContainer;
import com.sopra.ntts.model.NetworkStandard.MessageTypeEnum.MessageTypeEnum;

import java.util.Date;
import java.util.UUID;

/**
 * Message send by the client to notify the server about all the chosen operation.
 *
 * @author Standardisierungsdokument & Sedat Qaja
 */
public class GameOperationMessage extends MessageContainer {
    Operation operation;

  public GameOperationMessage(UUID playerId, MessageTypeEnum type, Date creationDate, String debugMessage, Operation operation) {
    super(playerId, MessageTypeEnum.GAME_OPERATION, creationDate, debugMessage);
    this.operation = operation;
  }
}
